cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MetricsConsumerMetaTable.java

```diff
@@ -47,11 +46,13 @@ public MetricsConsumerMetaTable(MetricsTable metaTable) {
     metaTable.put(updates);
   }
 
-  public synchronized <T extends MetricsMetaKey> void save(T key, byte[] value) throws Exception {
-    SortedMap<byte[], SortedMap<byte[], byte[]>> updates = Maps.newTreeMap(Bytes.BYTES_COMPARATOR);
-    SortedMap<byte[], byte[]> map = new TreeMap<>(Bytes.BYTES_COMPARATOR);
-    map.put(MESSAGE_ID_COLUMN, value);
-    updates.put(key.getKey(), map);
+  public <T extends MetricsMetaKey> void saveMessageIds(Map<T, byte[]> messageIds) throws Exception {
```

Comment:
lets keep it as synchronized and open jira to fix it later as all other methods in this class are synchronized.